pyart/util/columnsect.py

```diff
@@ -0,0 +1,294 @@
+                                           edges=True)
+    # Calculate distance from the x,y coordinates to target
+    rhidis = np.sqrt((rhi_x**2) + (rhi_y**2)) * np.sign(rhi_z)
+    for i in range(len(ray)):
+        tar_gate = np.argmin(abs(rhidis[i, 1:] - (dis)))
+        for key in moment:
+            moment[key].append(radar.fields[key]['data'][ray[i], tar_gate])
+        # fill the meta dictionary for further analysis
+        meta['xgate'].append(rhi_x[i, tar_gate])
+        meta['ygate'].append(rhi_y[i, tar_gate])
+        meta['zgate'].append(rhi_z[i, tar_gate])
+    # Combine the meta and moment data
+    column = dict(meta)
+    column.update(moment)
+
+    return column
+
+
+def get_column_rays(radar, azimuth):
+
+    """
+    Given the location (in lat,lon) of a target, return the rays that
+    correspond to radar column above the target.
+
+    Parameters
+    ----------
+    radar    : Radar Object
+               Py-ART Radar Object from which distance to the target, along
+               with gates above the target, will be calculated.
+    azimuth  : float,int
+               forward azimuth angle from radar to target in degrees.
+
+    Returns
+    -------
+    nrays : List
+            radar ray indices that correspond to the column above a
+            target location
+    """
+    if isinstance(azimuth, int) or isinstance(azimuth, float) is True:
+        if (azimuth <= 0) or (azimuth >= 360):
+            raise ValueError("azimuth not valid (not between 0-360 degrees)")
+    else:
+        raise TypeError("radar longitude type not valid")
+    # define a list to hold the valid rays
+    rays = []
+    # check to see which radar scan
+    if radar.scan_type == 'rhi':
+        for i in range(radar.sweep_number['data'].shape[0]):
+            nstart = radar.sweep_start_ray_index['data'][i]
+            nstop = radar.sweep_end_ray_index['data'][i]
+            counter = 0
+            for j in range(nstart, nstop):
+                if (abs(radar.azimuth['data'][nstart+counter] -
+                        azimuth) < 1):
+                    rays.append(nstart+counter)
+                counter += 1
+    else:
+        # taken from pyart.graph.RadarDisplay.get_azimuth_rhi_data_x_y_z
+        for sweep in radar.iter_slice():
+            sweep_azi = radar.azimuth['data'][sweep]
+            nray = np.argmin(np.abs(sweep_azi - azimuth))
+            rays.append(nray + sweep.start)
+    # make sure rays were found
+    if len(rays) == 0:
+        raise ValueError("No rays were found between azimuth and target")
+
+    return rays
+
+
+def check_lat(lat):
+
+    """
+    Function to check if input latitude is valid for type and value
+    """
+    if (isinstance(lat, int) or isinstance(lat, float) or
+            isinstance(lat, np.floating)) is True:
+        if (lat <= -90) or (lat >= 90):
+            raise ValueError("latitude not valid")
+    else:
+        raise TypeError("latitude type not valid")
```

Comment:
Same thing here - maybe mention that lat needs to be an int or float? Does this support a list of lats? Maybe add a check for that.